src/main/java/com/pashonokk/dvdrental/service/AddressService.java

```diff
@@ -0,0 +1,91 @@
+package com.pashonokk.dvdrental.service;
+
+import com.pashonokk.dvdrental.dto.AddressDto;
+import com.pashonokk.dvdrental.dto.AddressSavingDto;
+import com.pashonokk.dvdrental.entity.Address;
+import com.pashonokk.dvdrental.entity.Customer;
+import com.pashonokk.dvdrental.exception.CustomerNotFoundException;
+import com.pashonokk.dvdrental.mapper.AddressMapper;
+import com.pashonokk.dvdrental.mapper.AddressSavingMapper;
+import com.pashonokk.dvdrental.repository.AddressRepository;
+import com.pashonokk.dvdrental.repository.CustomerRepository;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class AddressService {
+    private final AddressRepository addressRepository;
+    private final CustomerRepository customerRepository;
+    private final AddressSavingMapper addressSavingMapper;
+    private final AddressMapper addressMapper;
+    private final Logger log = LoggerFactory.getLogger(AddressService.class);
+
+
+    @Transactional
+    public void addAddressToCustomer(AddressSavingDto addressSavingDto) {
+        Customer customer = customerRepository.findByIdWithAddress(addressSavingDto.getCustomerId())
+                .orElseThrow(() -> new CustomerNotFoundException("Customer with id " + addressSavingDto.getCustomerId() + " doesn`t exist"));
+        if (customer.getAddress() != null) {
+            log.warn("Customer {} already has address", customer);
+            return;
+        }
+        Address address = addressSavingMapper.toEntity(addressSavingDto);
+        customer.addAddress(address);
+    }
+
+    @Transactional
+    public void deleteCustomersAddress(Long id) {
+        Customer customer = customerRepository.findByIdWithAddress(id).orElse(null);
+        if (customer == null) {
+            return;
+        }
+        customer.removeAddress(customer.getAddress());
```

Comment:
I suggest to use ifPresent method: 
`@Transactional
public void deleteCustomerAddress(Long id) {
    customerRepository.findById(id).ifPresent(customer -> {
        customer.removeAddress(customer.getAddress());
    });
}
`